Avoid groups initialization  if required params aren't defined

ityaptin opened the pull request:
In ceilometer collector we don't need a initialization of decoder,
encoder, input and output if amqp, influxdb or elasticsearch connections
are not defined.

Commit:
Avoid definition of empty ceilometer_collector config groups

In ceilometer collector we don't need a initialization of decoder,
encoder, input and output if amqp, influxdb or elasticsearch connections
are not defined.

cloudlab-autobuild commented:
Can one of the admins verify this patch?

ityaptin commented:
@elemoine, @simonpasquier, @swann, @akomarek: Little fix for ceilometer_collector related configs

elemoine reviewed (changes requested):
@ityaptin would you mind doing the same for the other services (log_collector, etc.)? You can do that with additional commits.

Commit:
Avoid definition of empty stacklight heka config groups

Configuration groups as decoder, input, output or encoder shouldn't
be initialized without required params.

elemoine reviewed (changes requested):
@dkalashnik reported that this breaks the "heka" state.

dkalashnik reviewed (changes requested):
Failing state.sls for heka

Commit:
Fix behavior of heka_alarming.py for None service_data

Currently if service data has a key with None value:
`{a: None}`, function `grains_for_mine` raises exception.
It's incorrect. For right approach it should only pass this pair.

elemoine commented on heka/meta/heka.yml:
```diff
@@ -459,10 +468,12 @@ aggregator:
       full_action: drop
+  {%- endif %}
 {%- endif %}
+{%- if ceilometer_collector.influxdb_host is defined or ceilometer_collector.elasticsearch_host is defined or ceilometer_collector.amqp_host is defined %}
```
Please remove this.

@ityaptin and I discussed this on Slack, and we decided to make `heka_alarming.py:grains_for_mine` more robust.

elemoine reviewed (approved):
Thanks for finding and fixing the bug in `heka_alarming.py`.
